### ISSUE TITLE
Speed up processing

### ISSUE THREAD

**Issue by EarthScientist** [julienschroder/DOD_WRF#1]
The way the codebase was structured it is taking extremely long to run the data extractions.  This needs to be updated and made much faster.

**Comment by EarthScientist** [comment]
@julienschroder I have a version of the code that speeds the processing up substantially.  It is now taking ~3 mins to run across all of the `gfdl` files (both `hist` and `rcp85`).  

The update is a simple restructuring and cleanup of the existing `extraction.py` script and can be easily integrated into the codebase.  I would push the changes to a new branch and submit a pull request, but I will first need you to open up access to the repository for my user.

**Comment by julienschroder** [comment]
@EarthScientist cool! Do share :D. I will do that!

**Comment by EarthScientist** [comment]
cool PR has been sent.  Let me know if it doesnt fit your need or if you
just dont like the style of the code.  The essence of what you wanted to do
was correct, and has been preserved.

Sorry if I have overstepped my bounds here on code, but I got into it a bit
and this is the result.  :)  I also messed around with the `dask` approach,
which is slower but can definitely be used once that library has matured a
bit.

Michael
